src/types.rs

```diff
@@ -532,6 +586,42 @@ impl SourceMap {
         }
     }
 
+    /// Given a location, name and minified source file resolve a minified
+    /// name to an original function name.
+    ///
+    /// This invokes some guesswork and requires access to the original minified
+    /// source.  This will not yield proper results for anonymous functions or
+    /// functions that do not have clear function names.  (For instance it's
+    /// recommended that dotted function names are not passed to this
+    /// function).
+    pub fn get_original_function_name(&self, line: u32, col: u32,
+                                      minified_name: &str, source: &str) -> Option<&str> {
+        // fast way out if we are not looking up a valid javascript identifier
+        if !is_valid_javascript_identifier(minified_name) {
```

Comment:
This is just a quick check so that if someone looks for `Object.r.foo` or something we do not attempt to search through the file as we will never find a match anyways. Likewise `<unknown>` and stuff like this will return early.